lib/hooks.dart

```diff
@@ -101,3 +101,55 @@ StateHook<T> useState<T>(T initialValue) => StateHook(initialValue);
 ///
 /// Learn more: <https://reactjs.org/docs/hooks-reference.html#lazy-initial-state>.
 StateHook<T> useStateLazy<T>(T init()) => StateHook.lazy(init);
+
+/// Runs [sideEffect] after every completed render of a [DartFunctionComponent].
+///
+/// If [dependencies] are given, [sideEffect] will only run if one of the [dependencies] have changed.
+/// [sideEffect] may return a cleanup function that is run before the component unmounts or re-renders.
+///
+/// Note there are two rules for using Hooks (<https://reactjs.org/docs/hooks-rules.html>):
+///
+/// * Only call Hooks at the top level.
+/// * Only call Hooks from inside a [DartFunctionComponent].
+///
+/// __Example__:
+///
+/// ```
+/// UseEffectTestComponent(Map props) {
+///   final count = useState(1);
+///   final evenOdd = useState('even');
+///
+///   useEffect(() {
+///     if (count.value % 2 == 0) {
+///       evenOdd.set('even');
+///     } else {
+///       evenOdd.set('odd');
+///     }
+///     return () {
+///       print('count is changing...');
+///     };
+///   }, [count.value]);
```

Comment:
this isn't really required.
```suggestion
///   });
```

Reply:
I added that because the side effect is updating `evenOdd.value`. Without the dependency, every time `count.value` changes, the side effect will run twice (once because `count.value` changed and again because the side effect changed `evenOdd.value`). If the side effect did something like increment `evenOdd.value`, it would run in an infinite loop unless it had this dependency stopping it from running when `evenOdd.value` changes.

Reply:
Ooh could we add this as a comment in the example?